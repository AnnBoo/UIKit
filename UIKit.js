import UIActionComponent from './components/UIActionComponent';
import UIActionSheet from './components/menus/UIActionSheet';
import UIActionText from './components/text/UIActionText';
import UIAlertView from './components/popup/UIAlertView';
import UIAmountInput from './components/input/UIAmountInput';
import UIBackgroundView from './components/products/UIBackgroundView';
import UIBadge from './components/design/UIBadge';
import UIBottomBar from './components/products/UIBottomBar';
import UIButton from './components/buttons/UIButton';
import UICheckboxItem from './components/buttons/UICheckboxItem';
import UICard from './components/gram.scan/UICard';
import UIColor from './helpers/UIColor';
import UICompatibilityView from './helpers/UICompatibilityView';
import UIComponent from './components/UIComponent';
import UIConstant from './helpers/UIConstant';
import UIController from './controllers/UIController';
import UICustomSheet from './components/menus/UICustomSheet';
import UIDateInput from './components/input/UIDateInput';
import UIDetailsInput from './components/input/UIDetailsInput';
import UIDetailsToggle from './components/buttons/UIDetailsToggle';
import UIDetailsView from './components/text/UIDetailsView';
import UIDetailsTable from './components/gram.scan/UIDetailsTable';
import UIDevice from './helpers/UIDevice';
import UIDialogController from './controllers/UIDialogController';
import UIDot from './components/design/UIDot';
import UIDropdownAlert from './components/popup/UIDropdownAlert';
import UIEmailInput from './components/input/UIEmailInput';
import UIEventHelper from './helpers/UIEventHelper';
import UIFlashMessage from './helpers/UIFlashMessage';
import UIFont from './helpers/UIFont';
import UITextStyle from './helpers/UITextStyle';
import UIFunction from './helpers/UIFunction';
import UIImage from './components/images/UIImage';
import UIImageButton from './components/buttons/UIImageButton';
import UIImageView from './components/images/UIImageView';
import UILayoutManager from './helpers/UILayoutManager';
import UILinkInput from './components/input/UILinkInput';
import UIListHeader from './components/text/UIListHeader';
import UILoadMoreButton from './components/buttons/UILoadMoreButton';
import UILocalized from './helpers/UILocalized';
import UIMenuView from './components/menus/UIMenuView';
import UIMenuBackground from './helpers/UIMenuBackground';
import UIModalController from './controllers/UIModalController';
import UINavigationBackButton from './components/navigation/UINavigationBackButton';
import UINavigationBar from './components/navigation/UINavigationBar';
import UINavigator from './helpers/UINavigator';
import UINetworkStatus from './components/notifications/UINetworkStatus';
import UINotice from './components/notifications/UINotice';
import UIPasswordPrompt from './components/popup/UIPasswordPrompt';
import UIPhoneInput from './components/input/UIPhoneInput';
import UIProfilePhoto from './components/profile/UIProfilePhoto';
import UIProfileInitials from './components/profile/UIProfileInitials';
import UIProfileView from './components/profile/UIProfileView';
import UIPureComponent from './components/UIPureComponent';
import UIRadioButtonList from './components/buttons/UIRadioButtonList';
import UISearchBar from './components/input/UISearchBar';
import UISearchField from './components/gram.scan/UISearchField';
import UISectionHeader from './components/text/UISectionHeader';
import UISeparator from './components/design/UISeparator';
import UIShareManager from './helpers/UIShareManager';
import UISlider from './components/gram.scan/UISlider';
import UISpinnerOverlay from './components/UISpinnerOverlay';
import UISplitViewController from './controllers/UISplitViewController';
import UIStepBar from './components/navigation/UIStepBar';
import UIStubPage from './components/products/UIStubPage';
import UIStyle from './helpers/UIStyle';
import UISubtitle from './components/gram.scan/UISubtitle';
import UITabView from './components/menus/UITabView';
import UITextButton from './components/buttons/UITextButton';
import UITextInput from './components/input/UITextInput';
import UIToastMessage from './components/notifications/UIToastMessage';
import UITokenCell from './components/gram.scan/UITokenCell';
import UIToggle from './components/buttons/UIToggle';
import UITopBar from './components/gram.scan/UITopBar';
import UITooltip from './components/notifications/UITooltip';
import UIUserAgent from './helpers/UIUserAgent';

// Types
import type { CreateNavigationOptions, ReactNavigation } from './components/navigation/UINavigationBar';
import type { NavigationProps } from './helpers/UINavigator';
import type { PointerEvents, PositionObject } from './types';
import type { NavigationMenuList } from './components/gram.scan/UITopBar';
import type { ContentInset } from './controllers/UIController';
import type { DetailsList } from './components/gram.scan/UIDetailsTable';

export {
    UIActionComponent,
    UIActionSheet,
    UIActionText,
    UIAlertView,
    UIAmountInput,
    UIBadge,
    UIBackgroundView,
    UIBottomBar,
    UIButton,
    UICheckboxItem,
    UICard,
    UIColor,
    UICompatibilityView,
    UIComponent,
    UIConstant,
    UIController,
    UICustomSheet,
    UIDateInput,
    UIDetailsInput,
    UIDetailsToggle,
    UIDetailsView,
    UIDetailsTable,
    UIDevice,
    UIDialogController,
    UIDot,
    UIDropdownAlert,
    UIEmailInput,
    UIEventHelper,
    UIFlashMessage,
    UIFont,
    UITextStyle,
    UIFunction,
    UIImage,
    UIImageButton,
    UIImageView,
    UILayoutManager,
    UILinkInput,
    UIListHeader,
    UILoadMoreButton,
    UILocalized,
    UIMenuView,
    UIMenuBackground,
    UIModalController,
    UINavigationBackButton,
    UINavigationBar,
    UINavigator,
    UINetworkStatus,
    UINotice,
    UIPasswordPrompt,
    UIPhoneInput,
    UIProfilePhoto,
    UIProfileInitials,
    UIProfileView,
    UIPureComponent,
    UIRadioButtonList,
    UISearchBar,
    UISearchField,
    UISectionHeader,
    UISeparator,
    UIShareManager,
    UISlider,
    UISpinnerOverlay,
    UISplitViewController,
    UIStepBar,
    UIStubPage,
    UIStyle,
    UISubtitle,
    UITabView,
    UITextButton,
    UITextInput,
    UIToastMessage,
    UITokenCell,
    UITopBar,
    UIToggle,
    UITooltip,
    UIUserAgent,
};

export type {
    CreateNavigationOptions,
    NavigationMenuList,
    NavigationProps,
    PointerEvents,
    PositionObject,
    ReactNavigation,
    ContentInset,
    DetailsList,
};
