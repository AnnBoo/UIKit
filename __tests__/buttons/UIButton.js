import React from 'react';
import {UIButton} from '../../UIKit';
import { render } from '@testing-library/react-native';

test('Check UI button snap', () => {
  const { baseElement } = render(<UIButton />);
  expect(baseElement).toMatchSnapshot();
});
