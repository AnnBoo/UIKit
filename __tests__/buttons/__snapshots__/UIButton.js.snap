// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check UI button snap 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableWithoutFeedback
      disabled={true}
      testID="uiButton"
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
              "paddingHorizontal": 16,
            },
            Object {
              "backgroundColor": "#0088CC",
            },
            null,
            Object {
              "borderRadius": 4,
            },
            Object {
              "height": 48,
            },
            null,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "SFProText-Medium",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                },
                Object {
                  "color": "#FFFFFF",
                },
                null,
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </TouchableWithoutFeedback>
  </View>
</View>
`;
